File: 프로그래머스/2/68645. 삼각 달팽이/삼각 달팽이.py
def solution(n):
    cntarr = [[1 for i in range(1,j+1)] for j in range(1,n+1)] #그냥 생성
    
    x,y = -1,0
    cnt = 0
    for i in range(n):
        #i는 단순한 횟수, 숫자가 쓰지 않음
        for j in range(i,n):
            
            if i %3 == 0 :
                x+=1
            elif i%3 == 1 :
                y+=1 
            elif i%3 == 2 :
                x-=1
                y-=1
                
            cnt+=1
            cntarr[x][y] = cnt
            
            
    answer = []
    for i in range(len(cntarr)) :
        for j in cntarr[i]:
            answer.append(j)
            
    return answer

        

File: 프로그래머스/2/68645. 삼각 달팽이/삼각 달팽이.java
class Solution {
    
    public int [] dx = {1,0,-1};
    public int [] dy = {0,1,-1};
    
    public int[] solution(int n) {
        if (n ==1 )
        {
            return new int[] {1};
        }
        int [][]narr = new int [n][n]; 
        
        int x= -1;
        int y = 0;
        //down , right, left위
        
        int cnt = n;
        int whatdo = 0; //할일
        int val = 1;
        for (int move = n; move >=1 ; move --)
        {
            for (int i = 0 ; i<move; i++)
            {
                x = x+dx[whatdo%3];
                y = y+dy[whatdo%3];
                
                narr[x][y] = val;
                val+=1;
                //한 행동에서 할일 
            }
            whatdo +=1;
        }
        
        int nsize = 0;
        for (int i = 1; i<= n ; i++)
        {
            nsize += i;
        }
        
        int [] answer = new int[nsize];
        int ncnt = 0;
        for (int i = 0 ; i<n ; i ++)
        {
            for (int j = 0 ; j<=i ; j++)
            {
                if (narr[i][j]!=0)
                {
                    answer[ncnt] = narr[i][j];
                    ncnt +=1;
                }
            }
        }

        return answer;
    }
}